packages/primitives-core/src/styled.js

```diff
@@ -54,6 +54,11 @@ export function createStyled(
 
         let stylesWithStyleProp = styles
         if (props.style) {
+          if (typeof props.style === 'function') {
```

Comment:
alternatively, we could resolve those with `theme` - the main problem was that those were resolved with `props` which didn't even match css prop signatures

I'm mostly interested in avoiding any other solutions from being breaking changes in v11 - we don't have to have a final solution for this problem right now, we only need to make sure that we can implement a final solution within v11 line. But also - I'm not super interested in working on any improvements here 🤷‍♂️ The community must step up if they want anything from the libs related to React Native.

Reply:
This still allows functions here doesn't it because I assume you can pass in an array with a function in it and it'll be called?

Why don't we make `newProps.style = [emotionStyles, props.style]`?

Reply:
so that way RN should just resolve on its own the array and we omit the `props.style` from being resolved by Emotion? that would be sweet!

Reply:
Yeah

Reply:
I've used your advice and changes the code accordingly. Had to update snapshots and I've decided to use actual web implementation of `react-primitives` in tests so things get tested in more e2e manner - with the previous mock array got forwarded to the native (JS)DOM's style prop which didn't look OK.